### PR TITLE
windows: frame build out

### PR DESCRIPTION
**Pull request by tvanderstad**
fixes https://github.com/lockbook/lockbook/issues/2294
fixes https://github.com/lockbook/lockbook/issues/2271
fixes https://github.com/lockbook/lockbook/issues/2269
fixes https://github.com/lockbook/lockbook/issues/2268
fixes https://github.com/lockbook/lockbook/issues/2273
fixes https://github.com/lockbook/lockbook/issues/2154

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -12,6 +12,7 @@ members = [
 	"server/server",
 	"clients/cli",
 	"clients/egui",
+	"clients/windows",
```
Comment (tvanderstad) [inline Cargo.toml:15]: adds windows client to project

File: clients/egui/src/account/mod.rs [hunk 2]
```diff
@@ -38,7 +38,7 @@ use self::workspace::Workspace;
 pub struct AccountScreen {
     ctx: egui::Context,
     settings: Arc<RwLock<Settings>>,
-    core: lb::Core,
+    pub core: lb::Core,
```
Comment (tvanderstad) [inline clients/egui/src/account/mod.rs:41]: windows client grabs egui app's core

File: clients/egui/src/lib.rs [hunk 3]
```diff
@@ -261,6 +261,8 @@ impl WgpuLockbook {
             alpha_mode: CompositeAlphaMode::Auto,
             view_formats: vec![],
         };
-        self.surface.configure(&self.device, &surface_config);
+        if surface_config.width * surface_config.height != 0 {
```
Comment (tvanderstad) [inline clients/egui/src/lib.rs:264]: fixes crash on minimize, pattern I found in other open source projects

File: clients/egui/src/main.rs [hunk 4]
```diff
@@ -21,17 +21,23 @@ fn main() {
...
         eframe::NativeOptions {
             drag_and_drop_support: true,
             maximized: settings.window_maximize,
             initial_window_size: Some(egui::vec2(1300.0, 800.0)),
-            icon_data: Some(eframe::IconData {
-                rgba: include_bytes!("../lockbook.ico").to_vec(),
-                width: 32,
-                height: 32,
-            }),
+            icon_data: Some(eframe::IconData { rgba: icon_bytes, width: 128, height: 128 }),
```
Comment (tvanderstad) [inline clients/egui/src/main.rs:40]: we had an 88x128 image or something and this said 32x32. I tried a 256x256 image and 256's here to match but it silently showed a placeholder icon. 128x128 works but showed up backwards and upside down and with wrong colors without the inexplicable and arbitrary transformation above.

File: clients/windows/src/input/file_drop.rs [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+#[derive(Clone, Debug)]
+pub enum Message {
+    DragEnter { object: Option<IDataObject>, state: MODIFIERKEYS_FLAGS, point: POINTL },
+    DragOver { state: MODIFIERKEYS_FLAGS, point: POINTL },
+    DragLeave,
+    Drop { object: Option<IDataObject>, state: MODIFIERKEYS_FLAGS, point: POINTL },
+}
+
+#[implement(IDropTarget)]
```
Comment (tvanderstad) [inline clients/windows/src/input/file_drop.rs:17]: game changer macro for implementing Win32 C++ interfaces

File: clients/windows/src/input/file_drop.rs [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+pub fn handle_drop(app: &mut WgpuLockbook, object: Option<IDataObject>) -> bool {
+    if let Some(object) = object {
+        let format_enumerator: IEnumFORMATETC = unsafe {
+            object
+                .EnumFormatEtc(DATADIR_GET.0 as _)
+                .expect("enumerate drop formats")
+        };
+        let mut rgelt = [FORMATETC::default(); 1];
+        loop {
```
Comment (tvanderstad) [inline clients/windows/src/input/file_drop.rs:65]: note: dropping multiple files doesn't seem to be working

File: clients/windows/src/input/file_drop.rs [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+
+            if tymed == TYMED_HGLOBAL {
+                let hglobal = unsafe { stgm.u.hGlobal };
+
+                // for unknown reasons, if I don't cast the HGLOBAL to an HDROP and query the file count, the next call to object.GetData fails
+                // (this applies even if the format isn't CF_HDROP)
+                let hdrop = HDROP(unsafe { std::mem::transmute(hglobal) });
+                let file_count = unsafe { DragQueryFileW(hdrop, 0xFFFFFFFF, None) };
+
+                if format == CF_HDROP {
```
Comment (tvanderstad) [inline clients/windows/src/input/file_drop.rs:98]: a couple missing variants of these guys mean we don't support most types of drops, but we do support dropping a file dragged from file explorer so we're doing alright

File: clients/egui/src/lib.rs (hunk 3 above)
Comment (Parth) [reply clients/egui/src/lib.rs:264]: image

**Review by Parth — APPROVED** [review]

**Comment by Parth** [comment]
+ Probably the integration shouldn't need to know anything about core or our application logic
+ need some changes to releaser & winstaller: need to build new app on windows, and also need to build for arm

**Pull request merged**